src/main.rs

```diff
@@ -73,15 +74,38 @@ fn remove(opt: cli::Opt, key: String) -> Result<()> {
     config.save(&opt.amber_yaml)
 }
 
-fn print(opt: cli::Opt) -> Result<()> {
+fn print(opt: cli::Opt, style: cli::PrintStyle) -> Result<()> {
     let config = config::Config::load(&opt.amber_yaml)?;
     let secret = config.load_secret_key()?;
     let pairs: Result<Vec<_>> = config.iter_secrets(&secret).collect();
     let mut pairs = pairs?;
     pairs.sort_by(|x, y| x.0.cmp(y.0));
-    pairs
-        .iter()
-        .for_each(|(key, value)| println!("export {}={:?}", key, value));
+
+    let to_objs = |p: Vec<(&String, String)>| {
+        p.iter()
+            .map(|(k, v)| {
+                let mut obj = HashMap::new();
```

Comment:
I'd recommend a helper data type to avoid the overhead of creating a `HashMap`. Also, instead of cloning the `String`s here, you can hold onto the reference to the key and move the value. This is overall what I had in mind:

```rust
#[derive(Serialize)]
struct KeyValue<'a> {
    key: &'a str,
    value: String,
}

impl<'a> From<(&'a String, String)> for KeyValue<'a> {
    fn from((key, value): (&'a String, String)) -> Self {
        KeyValue { key, value }
    }
}

fn to_objs<'a>(p: Vec<(&'a String, String)>) -> Vec<KeyValue<'a>> {
    p.into_iter().map(KeyValue::from).collect::<Vec<_>>()
}
```

There are more subtle approaches with lifetime management that don't require the moving, but this seems Good Enough™️.

Reply:
Just because it's fun to go over the top:

```rust
#[derive(Serialize)]
struct KeyValue<'a> {
    key: &'a str,
    value: &'a str,
}

impl<'a, K, V> From<&'a (K, V)> for KeyValue<'a>
where
    K: AsRef<str>,
    V: AsRef<str>,
{
    fn from((key, value): &'a (K, V)) -> Self {
        KeyValue {
            key: key.as_ref(),
            value: value.as_ref(),
        }
    }
}

fn to_objs<'a, K, V, I>(p: I) -> Vec<KeyValue<'a>>
where
    I: IntoIterator<Item = &'a (K, V)>,
    K: AsRef<str> + 'a,
    V: AsRef<str> + 'a,
{
    p.into_iter().map(KeyValue::from).collect::<Vec<_>>()
}
```

Or the slightly terser to avoid the `From` impl:

```rust
#[derive(Serialize)]
struct KeyValue<'a> {
    key: &'a str,
    value: &'a str,
}

fn to_objs<'a, K, V, I>(p: I) -> Vec<KeyValue<'a>>
where
    I: IntoIterator<Item = &'a (K, V)>,
    K: AsRef<str> + 'a,
    V: AsRef<str> + 'a,
{
    p.into_iter()
        .map(|(key, value)| KeyValue {
            key: key.as_ref(),
            value: value.as_ref(),
        })
        .collect::<Vec<_>>()
}
```

Reply:
Seems like I have to drop the lambda and use inner function, for the lifetime.

Reply:
Interesting, using inner function saves the lifetime that I cannot specify in lambda.

Reply:
Did not see your generalized reply. Good call.

Reply:
It may be possible to do this with a closure for some simpler versions of the type signature and explicit return types/lifetimes, but for cases of closures that aren't capturing variables anyway, I prefer `fn` items. If you want, you can put the item inside the `print` function instead.

Reply:
Seems like generalization is way more complex and pops some memory management details.... Have not got it through.